pkg/util/error.go

```diff
@@ -43,23 +44,18 @@ func IsNotFound(err error) bool {
 	return false
 }
 
-func IsConnectionRefusedOrUnauthorizedOrAskingForCredentials(err error) bool {
+func IsConnectionRefusedOrUnauthorized(err error) bool {
 	if err == nil {
 		return false
 	}
 
-	msg := strings.ToLower(err.Error())
-	if strings.Contains(msg, "connection refused") || strings.Contains(msg, "unauthorized") || strings.Contains(msg, "the server has asked for the client to provide credentials") {
-		return true
-	}
-
-	return false
+	return errors.Is(err, syscall.ECONNREFUSED) || apierrors.IsUnauthorized(err)
```

Comment:
@c-pius, it's nice that you find these helper functions to check the error type.

Since this check is quite important, we need to have e2e test cover the case, can you create a follow up issue to design an e2e test regarding this? If need any help, feel free reach me.

Reply:
> can you create a follow up issue to design an e2e test regarding this

✅  https://github.com/kyma-project/lifecycle-manager/issues/1310